packages/api-fetch/src/middlewares/nonce.js

```diff
@@ -4,11 +4,9 @@
  */
 function createNonceMiddleware( nonce ) {
 	/**
-	 * @param {import('../types').ApiFetchRequestProps} options
-	 * @param {(options: import('../types').ApiFetchRequestProps) => import('../types').ApiFetchRequestProps} next
-	 * @return {import('../types').ApiFetchRequestProps} The enhanced request.
+	 * @type {import('../types').ApiFetchMiddleware & { nonce: string }}
```

Comment:
I guess we could do a bit better with typing the `{ nonce: string }` part.

Reply:
What would you suggest? This is the only way I know how to type a function with extra properties on it other than using a new interface:

```ts
interface ApiFetchNonceMiddleware extends ApiFetchMiddleware {
    nonce: string;
}
```

but that's effectively the same as what I have inline here.

Reply:
I agree, that's not much better 👍  Let's keep as-is for now